### ISSUE TITLE
Handle decorative images as it relates to Alt Text

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#3935]
Currently Drupal core requires alternative text for media images in attempt to improve accessibility, but at the same time,

>By [requiring alt text], this forces screen readers to read alt text for every image that is placed on the site especially when they have no purpose.

- https://www.drupal.org/project/drupal/issues/2934405

The above issue is actually an interesting read. Particularly the comment from the a11y core maintainer:  https://www.drupal.org/project/drupal/issues/2934405#comment-13962080 . There is a [contrib module , decorative_image_widget](https://www.drupal.org/project/decorative_image_widget), that tries to solve this issue, but doesn't solve the contextual issue with image usage.

>Most Drupal 8 and newer sites are using core's Media and Media Library modules, which decouple images from the entity (e.g. a content type), allowing the same media to be used in many places on your site by reference.

>It's important to keep in mind that good alternative text varies depending on the context it is used. A content author should describe a picture of fireworks differently when displayed in an article about photography techniques vs an article about the event in general.

>As such, while this module works just fine with the Media module, it's best not to rely on it if your site uses the same image in multiple different contexts, as you won't be able to change the alt text per context.

- https://www.drupal.org/project/decorative_image_widget

By adopting this approach, we are encouraging the upload of duplicate images in order to provide contextual alternative text if used on the site in multiple places, we are also giving editors an easy out if they can't think of appropriate alternative text. That is until something like https://www.drupal.org/project/drupal/issues/3023807 (being able to have context-based meta) is available.

WYSIWYG placed media does provide optional contextual image captioning in addition to the global alternative text.

Another idea was to alter renders on a case by case basis (banners, etc.) where we know they are being used essentially as decorative images due to their background position and remove the alt attribute programmatically.

Technically the current workaround many use is `""` which I think outputs as `alt=""""`. We could alter based on finding those instances as a "secret menu" workaround.

From what I have been reading, the alt attribute must be included to meet HTML spec and that there are certain exceptions to it having a value but should otherwise be a blank `alt=""`

I'd be curious to know how much of a nuisance "bad" alt text is compared to being able to know what an image describes. Are there other ways we can help speed up accessible navigation without compromising an accessible "feature" in Drupal core.

Related success criterion: https://www.w3.org/WAI/WCAG22/Techniques/failures/F38.html
> Procedure
For any img element that is used for purely decorative content:
> 1. Check whether the element has no role attribute or has a role attribute value that is not presentation.
> 2. Check whether the element has no alt attribute or has an alt attribute with a value that is not null.
> Expected Results
If check 1 and 2 are true, this failure condition applies and content fails the Success Criterion.

**Comment by joewhitsitt** [comment]
Curious to get additional thoughts from @inspector508 on this.

**Comment by joewhitsitt** [comment]
Not an issue solver, but core has made per instance alternative text a thing now. It uses the required global alt text by default, but provides a text input to change in the context the image is placed. https://github.com/uiowa/uiowa/pull/4741#issuecomment-1041778026

**Comment by briand44** [comment]
Step 1: Allow empty alt text, so the HTML is presented as alt=“ “
Step 2: Role = Presentation

This relates to 1.1.1 Level A

**Comment by joewhitsitt** [comment]
Module aside, there is some recent discussion about decorative images, alt text and aria here: https://drupal.slack.com/archives/C2ANFUGGG/p1741147392736239

alt or alt="" marks it as ignored in the accessibility tree. role="presentation" has the same effect and is thus considered redundant.

**Comment by briand44** [comment]
Grooming conversation... Sounds like there is interest in evaluating the contrib space vs doing this ourselves if we can get feedback from TM. 

https://www.drupal.org/project/decorative_images  - may be redundant with role and empty alt?
https://www.drupal.org/project/decorative_image_widget - want to make sure per instance doesn't get overridden by global default alt text. - https://github.com/uiowa/uiowa/issues/3935#issuecomment-1043188833

Can we address this in all the places we generate programmatic image renders.

Discussed if we did tackle this ourselves there was some interest in keeping alt required and forcing folks to enter an `""` or something that we then can strip out.

Investigation into contrib sized as medium

**Comment by briand44** [comment]
Closing this as #8711 has been merged.
